### PR TITLE
[DO-NOT-MERGE]direct-volume: fix long encoded file path

### PR DESCRIPTION
**Pull request by ChengyuZhu6**
shorten the encoded path by using a hashing algorithm SHA-256 to create a fixed-length string, which will be much shorter than the base64 encoding of the entire path.

Fixes: #10284

### CODE REVIEW

File: src/tools/kata-ctl/src/ops/volume_ops.rs [hunk 1]
```diff
@@ -158,10 +158,17 @@ mod tests {
     use super::*;
     use kata_types::mount::DirectVolumeMountInfo;
     use serial_test::serial;
+    use sha2::{Digest, Sha256};
     use std::{collections::HashMap, fs, path::PathBuf};
     use tempfile::tempdir;
     use test_utils::skip_if_not_root;
 
+    fn sha256_hex_digest(input: &str) -> String {
```
Comment (Apokleos) [inline src/tools/kata-ctl/src/ops/volume_ops.rs:166]: IMO，this is a public method not only for `kata-ctl` also for `join_path`.

Comment (ChengyuZhu6) [reply src/tools/kata-ctl/src/ops/volume_ops.rs]: Done.

**Review by sprt — CHANGES_REQUESTED** [review]
As mentioned in https://github.com/kata-containers/kata-containers/issues/10284#issuecomment-2341177716 and discussed offline, the original issue seems to be coming from `csi-kata-directvolume` passing an entire path instead of a basename, as in the shim: https://github.com/kata-containers/kata-containers/blob/66dda378773e450d074b569ddb7887aacc568371/src/runtime/virtcontainers/kata_agent.go#L1822

I think we should rather modify directly `csi-kata-directvolume` since this interface change could impact custom genpolicy policies.

**Comment by ChengyuZhu6** [comment]
> As mentioned in [#10284 (comment)](https://github.com/kata-containers/kata-containers/issues/10284#issuecomment-2341177716) and discussed offline, the original issue seems to be coming from `csi-kata-directvolume` passing an entire path instead of a basename, as in the shim:
> 
> https://github.com/kata-containers/kata-containers/blob/66dda378773e450d074b569ddb7887aacc568371/src/runtime/virtcontainers/kata_agent.go#L1822
> 
> I think we should rather modify directly `csi-kata-directvolume` since this interface change could impact custom genpolicy policies.

Got it. I’ll update the code in csi-kata-directvolume and revert the changes in shim.

**Comment by ChengyuZhu6** [comment]
This PR cannot fix the issue. I'd like to close this PR and keep the issue https://github.com/kata-containers/kata-containers/issues/10284 open.

**Pull request closed without merging**